### ISSUE TITLE
Where can I find document of this?

### ISSUE THREAD

**Issue by heroboy** [semiexp/csugar#2]
I know https://cspsat.gitlab.io/sugar/
But what is "graph-active-vertices-connected"?

**Comment by semiexp** [comment]
I'm afraid I have not written any document... 

Syntax for `graph-active-vertices-connected` is as follows:

```
(graph-active-vertices-connected <num-vertices> <num-edges> <v[0]> ... <v[n-1]> <p[0]> <q[0]> <p[1]> <q[1]> ... <p[m-1]> <q[m-1]>)
```

Here, `n` is `num-vertices` and `m` is `num-edges`. `<v[i]>` is boolean expressions, and `<p[i]>` and `<q[i]>` are integer literals (complex expressions are not allowed). 

The semantics for this operator is, simply put, "in a fixed graph, *selected* vertices are *connected*".
More precisely, suppose we have an undirected graph $G$ with $n$ vertices and $m$ edges. 
Vertices are numbered by $0$, $1$, $\dots$, $n-1$ and the $i$-th edge connects vertices `p[i]` and `q[i]`.
`v[i]` is a boolean expression representing whether the vertex $i$ is *selected*.
Then all selected vertices (those whose corresponding `v[i]` is true) should be *connected*, that is, for any selected two vertices $u$ and $v$, there must be a path connecting $u$ and $v$ such that vertices on the path are all selected.
In other words, the induced subgraph of $G$ with the selected vertices must be connected (or empty).

**Comment by heroboy** [comment]
Thank you for your detailed explanation.
I wonder why you add this when original sugar doesn't have this.

**Comment by semiexp** [comment]
I added this because this will be useful for solving pencil puzzle problems (https://github.com/semiexp/cspuz).
